### DISCUSSION TITLE
Where to store API keys for desktop app

### DISCUSSION THREAD

**Discussion by giobiondani** [Mapsui/Mapsui#2795] [Q&A]
I am writing a desktop app that uses Mapsui.
Some of the maps available with Mapsui (such as Bing maps) require an API key. I don’t want to include  my API key in the code itself because it would not be secure.
Does anybody have any recommendations on how to handle this?

**Comment by pauldendulk** [comment]
This is a more general (not Mapsui specific) question about how to work with secrets in your app. I think you need to search on how to work with secrets for the platform you are working on. I would be interested to hear about your solution.

**Reply by giobiondani** [reply]
Yes you are right, this is a more general question.
I did some research, and the best way should be to create a proxy API whereby the user (app) would interact with my API and my API would interact with the map providers. So the API would just send the tiles to the app. However, this is well beyond what I'm able to do on a hobby project.
So, what I'm looking at for now is to have the users authenticate on my app, then the app contacts a simple API that I will create, which provides the key at runtime. It's not airtight, especially if somebody is able to somehow create a valid token and call my API with it, but I hope that it would be more work than it's worth for them. Also, it would allow me to easily change an API key if I see that it's being abused.
If anyone has any better suggestions I'm all ears!

**Reply by giobiondani** [reply]
For context, I am writing a Windows App SDK desktop app using WinUI 3.
